### PR TITLE
first draft of project charter

### LINKED ISSUE #246: Develop a Project Charter to record decisions about how we will collaborate on spack-stack
I suggest we record our decisions about how the project will work in a project charter.

I will prepare a first draft for discussion.

When we are happy with it I will add a link from the README.

### PR DESCRIPTION
**Pull request by edwardhartnett**
Fixes #246 

This is a very rough first draft.

The goal here is to get the rules of the road straight so we can all get what we need from spack-stack without stepping on each other's toes or holding back the schedules of other organizations.

Please comment in the review...

### CODE REVIEW

File: project_charter.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+New packages must have a working package configuration file in
+spack. If not present, it must be added - if present, it must be
+examined and tested for correctness. It must be maintained with each
+new release of the package. Some package files are quite simple,
+others are very complex and take more work.
+
+Developing and maintaining the spack package file for each package is
+the responsibility of the organization that has requested that
+package. In the case of packages used by all organizations (like
+netcdf-c), the work of maintaining the spack package file is shared.
```
Comment (kgerheiser) [inline project_charter.md:58]: We can mention the `maintainers` section of the package here

File: project_charter.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+### Code Managers
+
+Organization | Code Manager(s)
+-------------|----------------
+NOAA EMC | Kyle Gerheiser, Hang Lei, Ed Hartnett
+JCSDA | Dom Heinzeller
+
+## Project Process
+
+The spack-stack project will follow the proeesses outlined in this
```
Comment (aerorahul) [inline project_charter.md]:
```suggestion
The spack-stack project will follow the processes outlined in this
```

File: project_charter.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+Spack package files should be pushed to the spack main repository, not
+held within the spack-stack project. For quick development and
+execution, package files can be temporarily housed in the spack-stack
+repo, but all should be moved to the main spack repo as time permits.
+
+### Installing on HPC Systems
+
+The spack-stack stack must be installed on various NOAA and NCAR
+systems. Each stakeholder will designiate programmers to install on
```
Comment (aerorahul) [inline project_charter.md]:
```suggestion
systems. Each stakeholder will designate programmers to install on
```

File: project_charter.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+### Tags
+
+Sometimes tags are preferrable to releases. Tags can be added at any time. Tag process:
+* Add a new issue documenting the reason for the tag.
+* Tag the repo.
+
+### Documentation
+
+spack-stack is documented with its README.md and on readthedocs.
```
Comment (aerorahul) [inline project_charter.md]: May be add link to RTD?

**Review by aerorahul — COMMENTED** [review]
Looks good. 
A few typos and a suggestion for a link to RTD.

**Comment by climbfuji** [comment]
I'll take a look in a few minutes.

File: project_charter.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+All package configuration files used by spack-stack are maintained in
+the main spack repository, making them available for all
+users. spack-spack is a thin shell on spack which contains necessary
+configuration files for our HPC systems, and some utilities which aid
+in installing the large number of packages managed by spack-stack.
+
+## Stakeholders
+
+- NOAA Environmental Modeling Center (EMC)
+- The UCAR Joint Center for Satellite Data Assimilation (JCSDA)
```
Comment (climbfuji) [inline project_charter.md:24]: Any reason why there is a "The" in line 24 and in line 25, but not in 23?

File: project_charter.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+others are very complex and take more work.
+
+Developing and maintaining the spack package file for each package is
+the responsibility of the organization that has requested that
+package. In the case of packages used by all organizations (like
+netcdf-c), the work of maintaining the spack package file is shared.
+
+Spack package files should be pushed to the spack main repository, not
+held within the spack-stack project. For quick development and
+execution, package files can be temporarily housed in the spack-stack
```
Comment (climbfuji) [inline project_charter.md]: I would say "housed in the `jcsda_emc_spack_stack` branch in the spack fork used by spack-stack" or something like that.

File: project_charter.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+repo, but all should be moved to the main spack repo as time permits.
+
+### Installing on HPC Systems
+
+The spack-stack stack must be installed on various NOAA and NCAR
```
Comment (climbfuji) [inline project_charter.md]: Not just NOAA and NCAR. JCSDA needs it on S4 (University of Wisconsin), Discover (NASA), and then there will be several Xsede platforms (Frontera for example) in the future.

**Review by climbfuji — CHANGES_REQUESTED** [review]
This is a very good first version of the project charter. I have a few comments, otherwise this is good to go from my side. Thanks for writing this up.

**Review by Hang-Lei-NOAA — APPROVED** [review]
Both the draft and comments/edits looks good. We may continuously improve it in later releases.

File: project_charter.md (hunk 1 above)
Comment (edwardhartnett) [reply project_charter.md:58]: What do you want to say?

File: project_charter.md (hunk 4 above)
Comment (edwardhartnett) [reply project_charter.md]: Done.

File: project_charter.md (hunk 5 above)
Comment (edwardhartnett) [reply project_charter.md:24]: Fixed.

File: project_charter.md (hunk 6 above)
Comment (edwardhartnett) [reply project_charter.md]: I made some changes.

File: project_charter.md (hunk 7 above)
Comment (edwardhartnett) [reply project_charter.md]: Fixed.

File: project_charter.md [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+
+### Tags
+
+Sometimes tags are preferrable to releases. Tags can be added at any time. Tag process:
+* Add a new issue documenting the reason for the tag.
+* Tag the repo.
+
+### Documentation
+
+spack-stack is documented with its README.md and [Spack Documentation
```
Comment (aerorahul) [inline project_charter.md:91]: I was suggesting https://spack-stack.readthedocs.io/en/latest/#

Comment (edwardhartnett) [reply project_charter.md:91]: Fixed.

**Review by climbfuji — APPROVED** [review]

**Comment by edwardhartnett** [comment]
I've made some minor README changes as well. I will leave this review up for a few hours and then merge it if no one has any further comments...

File: README.md [hunk 9]
```diff
@@ -1,19 +1,55 @@
...
+Spack-stack is a collaborative effort between:
+* [NOAA Environmental Modeling Center (EMC)](https://www.emc.ncep.noaa.gov/emc_new.php)
+* [UCAR Joint Center for Satellite Data Assimilation (JCSDA)](https://www.jcsda.org/)
+* [Earth Prediction Innovation Center (EPIC)](https://epic.noaa.gov/).
 
-spack-stack is mainly a collection of Spack configuration files, but provides a Spack extension to simplify the installation process:
-- `spack stack create` is provided to copy common, site-specific, and application-specific configuration files into a coherent Spack environment and to create container recipes
-- `spack stack setup-meta-modules` creates compiler, MPI and Python meta-modules for a convenient setup of a user environment using modules (lua and tcl)
+Spack-stack is a fork of the [spack](https://github.com/spack/spack)
```
Comment (aerorahul) [inline README.md]: I don't think this is accurate.
The accurate statement would be:
`Spack-stack` contains a gitsubmodule of the fork of Spack.
Spack stack is a "recipe" repository much similar to: https://github.com/spack/spack-configs

Comment (climbfuji) [inline README.md]: Spack-stack includes a fork ? leverages a fork?

Comment (climbfuji) [reply README.md]: Or "Spack-stack is a thin layer around a fork of the ..." ?

**Review by climbfuji — COMMENTED** [review]
Just one more suggestion for the README, but it's ok to leave it as is as well.

File: README.md (hunk 9 above)
Comment (edwardhartnett) [reply README.md]: OK fixed

**Review by climbfuji — APPROVED** [review]

**Review by aerorahul — APPROVED** [review]
Looks good.
Thanks for incorporating suggestions.

**Pull request merged**